src/native/libs/System.Native/pal_console.c

```diff
@@ -39,6 +39,19 @@ int32_t SystemNative_GetWindowSize(WinSize* windowSize)
 #endif
 }
 
+int32_t SystemNative_SetWindowSize(WinSize* windowSize)
+{
+    assert(windowSize != NULL);
+
+#if HAVE_IOCTL && HAVE_TIOCSWINSZ
+    int error = ioctl(STDOUT_FILENO, TIOCSWINSZ, windowSize);
+    return error;
```

Comment:
> The [ioctl(2)](https://man7.org/linux/man-pages/man2/ioctl.2.html) system call returns 0 on success.  On error, it
       returns -1 and sets [errno](https://man7.org/linux/man-pages/man3/errno.3.html) to indicate the error.

nit: Calling it `error` here suggests that the sys-call does not set the error but returns it instead. You can just return the value returned by `ioctl` 
```suggestion
    return ioctl(STDOUT_FILENO, TIOCSWINSZ, windowSize);
```